test/cluster.go

```diff
@@ -17,24 +17,22 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
-func CreateCluster(t *testing.T, db *gorm.DB) *repository.Cluster {
-
+// CreateCluster returns a new cluster after saves it in the DB
+func CreateCluster(t *testing.T, db *gorm.DB) repository.Cluster {
 	cluster := NewCluster()
 	repo := repository.NewClusterRepository(db)
-
-	err := repo.Create(context.Background(), cluster)
+	err := repo.Create(context.Background(), &cluster)
 	require.NoError(t, err)
-
+	// verify that the cluster exists in the DB
 	cls, err := repo.Load(context.Background(), cluster.ClusterID)
 	require.NoError(t, err)
-
-	AssertEqualCluster(t, cluster, cls, true)
+	AssertEqualCluster(t, cluster, *cls, true)
 	return cluster
 }
 
 // NewCluster returns a new cluster with random values for all fields
-func NewCluster() *repository.Cluster {
-	return &repository.Cluster{
+func NewCluster() repository.Cluster {
```

Comment:
This looks bit odd/confusing. In Go, if we have a function which starts with `New`, it generally returns a pointer, but here we are not returning pointer.

Reply:
I see what you mean, but I really don't see why we need to return a pointer here, since there's no `error` to return if something wrong happened...